Handle compression element

gwen-lg opened the pull request:
Some tracks have compression, to be able to handle frame data correctly, we need to get compression information.

draft :
because after made a first version, I thought that the management of `ContentEncoding` could be _improved_ by use an enum with values for content encoding value, instead of a value for type, and two Option for values.
It's seems more the Rust way to me.
But it's a breaking change, to before to clean changes (rework commits, check doc, ...), I prefer as you @hasenbanck if you are ok, or if you prefer keep only the changes from the first commit.
Also: I will try to make a test file before the final version.

gwen-lg committed:
rework ContentEncoding variants management.

add an enum for manage ContentEncoding variants.
Use it instead of an Option for ContentEncryption.

gwen-lg committed:
Support Content Compression

- add element id : ContentCompression, ContentCompAlgo and ContentCompSettings.
- add ContentCompAlgo enum
- add struct ContentCompression to store compression algo + settings
- add compression in ContentEncodingValue::Compression variant

hasenbanck commented on src/enums.rs:
```diff
@@ -463,6 +463,44 @@ impl From<u64> for ContentEncodingType {
 
+/// The compression algorithm used.
+///
+/// <div class="warning">
```
Can you just add a `# Notes` here? No need for the custom div.

gwen-lg replied:
:ok:

hasenbanck commented on src/lib.rs:
```diff
@@ -922,13 +922,23 @@ impl MasteringMetadata {
+    Unknown,
+    /// Indicate than content is compressed.
+    Compression(ContentCompression),
+    /// Indicate than content is encrypted. And add information
```
I think you can remove "And add information" here.

gwen-lg replied:
:ok:

hasenbanck reviewed (commented):
In general it's fine to break the API, since we are not stable anyhow.

Is it not possibel to use compression AND encryption at the same time? That seems odd for me.

gwen-lg commented:
> In general it's fine to break the API, since we are not stable anyhow.

:ok_hand:  

> Is it not possibel to use compression AND encryption at the same time? That seems odd for me.

A `ContentEncoding` is dedicated to one transformation, but a track have a list of `ContentEncoding` witch is perfectly managed, and accessible with `content_encodings`. :smirk:

gwen-lg committed:
remove `encoding_type()` accessor in `ContentEncoding`

it's duplicates with `encoding()` who return a `ContentEncodingValue`.
The `ContentEncodingValue` as variants who indicate the type of encoding.
Keep `ContentEncodingType` only for internal management.

gwen-lg committed:
Add test for pgs subtitles with zlib compression

include new test file `tests/data/subtitles_pgs.mkv`

gwen-lg commented:
I also add a commit to [remove encoding_type() accessor in ContentEncoding](https://github.com/hasenbanck/matroska-demuxer/pull/23/commits/09f1eb221a465f7b8bf0a8514d3de35ac54add16) as it's duplicates with `encoding()` who return a `ContentEncodingValue`.
I you prefer, I can easily restore it, as it's last commit.

gwen-lg committed:
Add test for mkv with vobsub subtitle tracks
